VoiceAttackResponder/VoiceAttackPlugin.cs

```diff
@@ -310,93 +359,181 @@ private static void setJsonValues(ref dynamic vaProxy, string prefix, dynamic js
+                        case "initialize eddi":
+                            if (eddiInstance)
+                                vaProxy.WriteToLog("The EDDI plugin is fully operational.", "green");
+                            else
+                                VA_Init1(vaProxy);  // Attempt initialization again to see if it works this time...
+
+                            break;
+                        case "configuration":
+                        case "configurationminimize":
+                        case "configurationmaximize":
+                        case "configurationrestore":
+                        case "configurationclose":
+                            // Ignore any attempt to access the EDDI UI if VA
+                            // doesn't have the EDDI instance.
+                            if (eddiInstance)
+                                InvokeConfiguration(ref vaProxy);
+                            else
+                                vaProxy.WriteToLog("The EDDI plugin is not fully initialized.", "red");
+                            break;
+                        case "shutup":
+                            InvokeShutUp(ref vaProxy);
+                            break;
+                        case "setstate":
+                            InvokeSetState(ref vaProxy);
+                            break;
+                        case "disablespeechresponder":
+                            InvokeDisableSpeechResponder(ref vaProxy);
+                            break;
+                        case "enablespeechresponder":
+                            InvokeEnableSpeechResponder(ref vaProxy);
+                            break;
+                        case "setspeechresponderpersonality":
+                            InvokeSetSpeechResponderPersonality(ref vaProxy);
+                            break;
+                    }
                 }
-            }
-            catch (Exception e)
-            {
-                Logging.Error("Failed to invoke action " + vaProxy.Context, e);
-                vaProxy.WriteToLog("Failed to invoke action " + vaProxy.Context);
-            }
+                catch (Exception e)
+                {
+                    Logging.Error("Failed to invoke context " + vaProxy.Context, e);
+                    vaProxy.WriteToLog("Failed to invoke context " + vaProxy.Context, "red");
+                }
+            //}
         }
 
         public static void VA_StopCommand()
         {
         }
 
+        // Allow only one instance of the EDDI configuration UI
+        private static MainWindow configWindow = null;
+        //private static bool configWinRunning = false;
+        private static Thread configThread = null;
+
         private static void InvokeConfiguration(ref dynamic vaProxy)
         {
-            Thread thread = new Thread(() =>
+            string config = (string)vaProxy.Context;
+
+            if (null == configWindow && config != "configuration")
```

Comment:
Yoda syntax distorts readability and is not needed with modern compilers that warn about accidental use of `=` rather than `==`.

Note how it is only used in the first half of the `&&`. Full Yoda syntax would be `if (null == configWindow && "configuration" != config)`.

Note how much more readable the alternative is:
`if (configWindow == null && config != "configuration")`

In any case, if you remembered to use Yoda syntax, you therefore would have remembered to use `==` anyway. 

It is a silly fad that modern compilers have let us do away with, along with "Hungarian" variable prefixes.